README.md
vkDevastator
============

Набор скриптов позволяющий найти и удалить всю* вашу активность** из социальной сети ВКонтакте.

(*) поиск происходит путем полного перебора групп и друзей, если вас ничто не связывает с пользователем или группой,
то найти вашу активность не представляется возможным.

(**) на данный момент поддерживаются: фотографии, комментарии к фотографиям, видео, комментарии к видео, заметки, 
комментарии к заметкам, обсуждения, комментарии к обсуждениям, записи на стене и комментарии к записям на стене.

Рекомендации
------------
* Если вы совершили некоторую активность в группе или в профиле пользователя - не удаляйте их (группу, пользователя).
В дальнейшем это облегчит поиск и удаление вашей активности.

Установка и запуск
------------------
Для загрузки скрипта вам потребуется git а так же python 2.x для запуска

В Debian/Ubuntu поставить их можно следующим образом:
    
    sudo apt-get install git python

Загрузка и установка скрипта:

    cd ~
    git clone https://github.com/AlekseyDurachenko/vkDevastator vkDevastator
    
Запуск скрипта

    cd ~/vkDevastator/src
    #python ./vksearchactivities.py --access-token XXXYYYZZZ --target-id ID --state-file state.txt --activities-file activities.txt --activities-detail-file detail.txt
    #vkdeleteactivities.py --access-token XXXYYYZZZ --activities-file activities.txt
    
Получение ACCESS TOKEN
----------------------
* Во-первых, вам необходимо зарегистрировать новое приложение типа **Standalone-приложение** 
(http://vk.com/editapp?act=create), или использовать уже имеющееся. 
Так или иначе вам потребуется его уникальный **APP_ID**. 
* После того, как вам стал известен ваш **APP_ID** следует пройти OAuth авторизацию для получения access_token.
Для этого перейдите по ссылке https://oauth.vk.com/authorize?client_id=APP_ID&scope=friends,photos,audio,video,docs,notes,pages,status,wall,groups,offline&display=page&redirect_uri=https://oauth.vk.com/blank.html&response_type=token& 
и авторизируйтесь(вместо **APP_ID** поставьте ID вашего приложения). 
В случае успешной авторизации в адресной строке вы увидите access_token=XXXYYYZZZ. 
Запомните это значение в дальнейшем оно вам потребуется неоднократно.
    
    **Примечание! Перед получением access_token следуе выйти из ВКонтакте или открыть приватную вкладку.**

Поиск вашей активности (vksearchactivities.py)
----------------------------------------------

    == vksearchactivities.py - v.0.1.1  ==
    Usage: 
        vksearchactivities.py --access-token <> --target-id <> --state-file <> --activities-file <> --activities-detail-file <>
    
        --log-file <>   Path to the log file
        --search-user-depth  N   (default 1) The users search depth
        --search-group-depth N   (default 1) The groups search depth
        --custom-user-ids    N   The list of custom user_id splitted by ","
        --custom-group-ids   N   The list of custom group_id splitted by "," (positive values)
        --disable-scan-friends   Ignore the friends
        --disable-scan-followers  Ignore the followers
        --disable-scan-user-subscriptions   Ignore the subscriptions to users
        --disable-scan-group-subscriptions  Ignore the subscriptions to groups
        --show-api-queries   Show the API queries
        --show-api-errors    Show the API errors
        --limit-member-count   Limit the maximum member count of the group
        --scan-time-limit    N   limit the time of the scanning of user(group), in minutes
        --enable-scan-himself    Don't ignore himself
        --disable-scan-walls
        --disable-scan-photos
        --disable-scan-photocomments
        --disable-scan-videos
        --disable-scan-topics

Данный скрипт используется для поиска всей вашей активности. В простейшем случае запуск осуществляется следующим образом:

    python ./vksearchactivities.py --access-token XXXYYYZZZ --target-id ID --state-file state.txt --activities-file activities.txt --activities-detail-file detail.txt

Подставьте в этот скрипт вместо XXXYYYZZZ ваш access_token а вместо ID ваш идентификатор пользователя ВКонтакте
(узнать его можно на странице http://vk.com/settings в поле "Номер страницы"). 
В результате поиск вашей активности будет произведен во всех ваших группах, друзьях, подписчиках и подписках. 

* Ознакомиться с результатами поиска можно в файле detail.txt
* Эти же результаты, но в машинно читаемом виде хранятся в файле activities.txt
* В файле state.txt хранятся идентификаторы пользователь и групп, которые уже были проверены на наличие вашей активности.
Если вы хотите произвести поиск заново - просто удалите этот файл.
* Скрипт можно остановить с помощью CTRL+C. Когда вы запустите его вновь поиск продолжится с места остановки.

Иногда бывает недостаточно проверки только своих групп и своих друзей. На этот случай предоставляется возможность
искать группы друзей, друзей друзй, группы друзей друзей и т.п. За это отвечают параметры --search-user-depth N,
--search-group-depth N. Где N - глубина поиска. "0" означает, что проверка друзей/групп не производится. "1" означет
что поиск производится только в своих группах/друзьях. "2" означает, что поиск друзей/групп производится у друзей 
и так далее.

    python ./vksearchactivities.py --access-token XXXYYYZZZ --target-id ID --state-file state.txt --activities-file activities.txt --activities-detail-file detail.txt -search-user-depth 1 --search-group-depth 2

В данном случае поиск будет произведен в ваших группах, друзьях, подписчиках, подписках а так же в 
группах друзей, подписчиков, подписках.

Если в группе слишком много участников, то это, как правило, означает большое кол-во контента, а следовательно
и поиск вашей активности может затянуться надолго. Чтобы этого избежать рекомендуется исключить поиск в группах
с кол-во участников превышающим определенное число. Сделать это можно при помощи --limit-member-count N,
где N максимальное кол-во участников в группе.

    python ./vksearchactivities.py --access-token XXXYYYZZZ --target-id ID --state-file state.txt --activities-file activities.txt --activities-detail-file detail.txt --search-user-depth 1 --search-group-depth 2 --limit-member-count 1000

Бывает так, что некоторые пользователи или группы имеют слишком большую стену,
слишком много фотографий и т.п., что приводит к очень долгому сканированию
данного пользователя(группы). Чтобы этого избежать, достаточно задать 
колюч --scan-time-limit N, где N - кол-во минут, которое допускается потратить
на одного пользователя(группу). Если сканирование провести не удалось, то
пользователь(группа) пропускается а ее id помечается как "просканированная",
и при повторном запуске не будет сканироваться повторно.

    ./vksearchactivities.py --access-token XXXYYYZZZ --target-id ID --state-file state.txt --activities-file activities.txt --activities-detail-file detail.txt --search-user-depth 0 --search-group-depth 0 --enable-scan-himself --disable-scan-photos --disable-scan-photocomments --disable-scan-videos --disable-scan-topics
    
    Таким образом вы можете очистить свою стену. (не забудьте, что после данной команды потребуется так же выполнить vkdeleteactivities.py)

Удаление вашей активности (vkdeleteactivities.py)
-------------------------------------------------

    == vkdeleteactivities.py - v.0.1.1  ==
    Usage: 
        vkdeleteactivities.py --access-token <> --activities-file <>

После того, как вы произвели поиск вашей активности и получили файл activities.txt вам следует запустить:

    vkdeleteactivities.py --access-token XXXYYYZZZ --activities-file activities.txt
    
Подставьте в этот скрипт вместо XXXYYYZZZ ваш access_token.

***ВНИМАНИЕ! После выполнение этой команды восстановить вашу активность хоть и будет возможно,
но готового скрипта я пока еще не написал, и не планирую =)***

Заключение
----------

Поздравляем! Теперь все, что вы когда-либо писали (конечно, из того, что получилось отыскать), более недоступно
пользователям социальной сети. Впредь будьте бдительны и осторожны в ваших высказываниях. Ведь
слово, как говорится, не воробей...

P.S.: предложения, пожелания, ошибки - в багтрекер.


CHANGELOG
ChangeLog
=========

v0.1.0
------

* Первая версия

v0.1.1
------

* Добавлен аргумент --limit-member-count
* bug fixes

next
----

* добавлен аргумент --scan-time-limit
* добавлен аргумент --enable-scan-himself
* добавлен аргумент --disable-scan-walls
* добавлен аргумент --disable-scan-photos
* добавлен аргумент --disable-scan-photocomments
* добавлен аргумент --disable-scan-videos
* добавлен аргумент --disable-scan-topics


src/vkdelete.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# Copyright 2013, Durachenko Aleksey V. <durachenko.aleksey@gmail.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
# The remove order:
# * POST_COMMENT owner_id, post_id, comment_id
# * POST owner_id, post_id
# * PHOTO_COMMENT owner_id, comment_id
# * PHOTO owner_id, photo_id
# * VIDEO_COMMENT owner_id, video_id, comment_id
# * VIDEO owner_id, topic_id
# * TOPIC_COMMENT group_id, topic_id, comment_id
# * TOPIC group_id, topic_id
# * NOTE_COMMENT user_id, note_id, comment_id
# * NOTE user_id, note_id
import urllib2, json, time, getopt, sys, socket

def executeQuery(query):
    localQuery = query
    while True:
        print localQuery
        try:
            f = urllib2.urlopen(localQuery)
            data = json.load(f)
            f.close()
        except:
            time.sleep(2)
            continue
        
        if "error" in data:
            print "ERROR: ", data
            # captcha needed
            if data["error"]["error_code"] == 14:
                captcha = raw_input("Enter a captcha code:")
                localQuery = query + "&captcha_sid=%d&captcha_key=%s" % (int(data['error']['captcha_sid']), captcha)
                continue
            if data["error"]["error_code"] == 15:
                return False
            # user was deleted or banned
            if data["error"]["error_code"] == 18:
                return False
            # too many queryes per second
            if data["error"]["error_code"] == 6:
                time.sleep(0.5)
                continue
            else:
                #print "ERROR: ", data
                return False
        # end if
    
        return True
    # end while        

def removeWallComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/wall.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removeWall(access_token, owner_id, post_id):
    return executeQuery("https://api.vk.com/method/wall.delete?v=5.2&access_token=%s&owner_id=%d&post_id=%d" % (access_token, int(owner_id), int(post_id)))

def removePhotoComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/photos.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removePhoto(access_token, owner_id, photo_id):
    return executeQuery("https://api.vk.com/method/photos.delete?v=5.2&access_token=%s&owner_id=%d&photo_id=%d" % (access_token, int(owner_id), int(photo_id)))

def removeVideoComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/video.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removeVideo(access_token, owner_id, video_id):
    return executeQuery("https://api.vk.com/method/video.delete?v=5.2&access_token=%s&owner_id=%d&video_id=%d" % (access_token, int(owner_id), int(video_id)))

def removeTopicComment(access_token, group_id, topic_id, comment_id):
    return executeQuery("https://api.vk.com/method/board.deleteComment?v=5.2&access_token=%s&group_id=%d&topic_id=%d&comment_id=%d" % (access_token, int(group_id), int(topic_id), int(comment_id)))

def removeTopic(access_token, group_id, topic_id):
    return executeQuery("https://api.vk.com/method/board.deleteTopic?v=5.2&access_token=%s&group_id=%d&topic_id=%d" % (access_token, int(group_id), int(topic_id)))

def removeNoteComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/notes.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removeNote(access_token, note_id):
    return executeQuery("https://api.vk.com/method/notes.delete?v=5.2&access_token=%s&note_id=%d" % (access_token, int(note_id)))

def showUsage():
    print "== vkdelete.py - v.0.1.0  =="
    print "Usage: "
    print "    vkdelete.py --access_token <> --activity-file <>"


access_token = None
activity_file = None

options, remainder = getopt.getopt(sys.argv[1:], 'o:v', ['access_token=', 'activity-file='])
for opt, arg in options:
    if opt == '--access_token':
        access_token = arg
    elif opt == '--activity-file':
        activity_file = arg

if access_token == None or activity_file == None:
    showUsage()
    exit(-1)

print "Access token :", access_token
print "Activty file :", activity_file

# vk can hold the connection for a long time
socket.setdefaulttimeout(3)

# read lines with activities
dbFile = open(activity_file, 'r')
lines = dbFile.read().split("\n")
lines = list(set(lines))

# remove POST_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'POST_COMMENT ' in line:
        [type_id, owner_id, post_id, comment_id] = line.split(" ")
        if removeWallComment(access_token, owner_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "POST_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove POST
total_count = delete_count = 0
for line in lines:
    if 'POST ' in line:
        [type_id, owner_id, post_id] = line.split(" ")
        if removeWall(access_token, owner_id, post_id) == True:
            delete_count += 1
        total_count += 1
print "POST: %d of %d was removed" % (delete_count, total_count)

# remove PHOTO_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'PHOTO_COMMENT ' in line:
        [type_id, owner_id, comment_id] = line.split(" ")
        if removePhotoComment(access_token, owner_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "PHOTO_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove PHOTO
total_count = delete_count = edit_count = 0
for line in lines:
    if 'PHOTO ' in line:
        [type_id, owner_id, photo_id] = line.split(" ")
        if removePhoto(access_token, owner_id, photo_id) == True:
            delete_count += 1
        total_count += 1
print "PHOTO: %d of %d was removed" % (delete_count, total_count)

# remove VIDEO_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'VIDEO_COMMENT ' in line:
        [type_id, owner_id, video_id, comment_id] = line.split(" ")
        if removeVideoComment(access_token, owner_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "VIDEO_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove VIDEO
total_count = delete_count = edit_count = 0
for line in lines:
    if 'VIDEO ' in line:
        [type_id, owner_id, video_id] = line.split(" ")
        if removeVideo(access_token, owner_id, video_id) == True:
            delete_count += 1
        total_count += 1
print "VIDEO: %d of %d was removed" % (delete_count, total_count)

# remove TOPIC_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'TOPIC_COMMENT ' in line:
        [type_id, group_id, topic_id, comment_id] = line.split(" ")
        if removeTopicComment(access_token, group_id, topic_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "TOPIC_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove TOPIC
total_count = delete_count = edit_count = 0
for line in lines:
    if 'TOPIC ' in line:
        [type_id, group_id, topic_id] = line.split(" ")
        if removeTopic(access_token, group_id, topic_id) == True:
            delete_count += 1
        total_count += 1
print "TOPIC: %d of %d was removed" % (delete_count, total_count)


# remove NOTE_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'NOTE_COMMENT ' in line:
        [type_id, user_id, note_id, comment_id] = line.split(" ")
        if removeNoteComment(access_token, user_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "NOTE_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove NOTE
total_count = delete_count = edit_count = 0
for line in lines:
    if 'NOTE ' in line:
        [type_id, user_id, note_id] = line.split(" ")
        if removeNote(access_token, note_id) == True:
            delete_count += 1
        total_count += 1
print "NOTE: %d of %d was removed" % (delete_count, total_count)


src/vkdeleteactivities.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
# Copyright 2013, Durachenko Aleksey V. <durachenko.aleksey@gmail.com>
#
# This program is free software: you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation, either version 3 of the License, or
# (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.
#
# The remove order:
# * POST_COMMENT owner_id, post_id, comment_id
# * POST owner_id, post_id
# * PHOTO_COMMENT owner_id, comment_id
# * PHOTO owner_id, photo_id
# * VIDEO_COMMENT owner_id, video_id, comment_id
# * VIDEO owner_id, topic_id
# * TOPIC_COMMENT group_id, topic_id, comment_id
# * TOPIC group_id, topic_id
# * NOTE_COMMENT user_id, note_id, comment_id
# * NOTE user_id, note_id
import urllib2, json, time, getopt, sys, socket

def executeQuery(query):
    localQuery = query
    while True:
        print localQuery
        try:
            f = urllib2.urlopen(localQuery)
            data = json.load(f)
            f.close()
        except:
            time.sleep(2)
            continue
        
        if "error" in data:
            print "ERROR: ", data
            # captcha needed
            if data["error"]["error_code"] == 14:
                captcha = raw_input("Enter a captcha code:")
                localQuery = query + "&captcha_sid=%d&captcha_key=%s" % (int(data['error']['captcha_sid']), captcha)
                continue
            if data["error"]["error_code"] == 15:
                return False
            # user was deleted or banned
            if data["error"]["error_code"] == 18:
                return False
            # too many queryes per second
            if data["error"]["error_code"] == 6:
                time.sleep(0.5)
                continue
            else:
                return False
        # end if
    
        return True
    # end while        

def removeWallComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/wall.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removeWall(access_token, owner_id, post_id):
    return executeQuery("https://api.vk.com/method/wall.delete?v=5.2&access_token=%s&owner_id=%d&post_id=%d" % (access_token, int(owner_id), int(post_id)))

def removePhotoComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/photos.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removePhoto(access_token, owner_id, photo_id):
    return executeQuery("https://api.vk.com/method/photos.delete?v=5.2&access_token=%s&owner_id=%d&photo_id=%d" % (access_token, int(owner_id), int(photo_id)))

def removeVideoComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/video.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removeVideo(access_token, owner_id, video_id):
    return executeQuery("https://api.vk.com/method/video.delete?v=5.2&access_token=%s&owner_id=%d&video_id=%d" % (access_token, int(owner_id), int(video_id)))

def removeTopicComment(access_token, group_id, topic_id, comment_id):
    return executeQuery("https://api.vk.com/method/board.deleteComment?v=5.2&access_token=%s&group_id=%d&topic_id=%d&comment_id=%d" % (access_token, int(group_id), int(topic_id), int(comment_id)))

def removeTopic(access_token, group_id, topic_id):
    return executeQuery("https://api.vk.com/method/board.deleteTopic?v=5.2&access_token=%s&group_id=%d&topic_id=%d" % (access_token, int(group_id), int(topic_id)))

def removeNoteComment(access_token, owner_id, comment_id):
    return executeQuery("https://api.vk.com/method/notes.deleteComment?v=5.2&access_token=%s&owner_id=%d&comment_id=%d" % (access_token, int(owner_id), int(comment_id)))

def removeNote(access_token, note_id):
    return executeQuery("https://api.vk.com/method/notes.delete?v=5.2&access_token=%s&note_id=%d" % (access_token, int(note_id)))

def showUsage():
    print "== vkdeleteactivities.py - v.0.1.1  =="
    print "Usage: "
    print "    vkdeleteactivities.py --access-token <> --activities-file <>"


access_token = None
activities_file = None

try:
    options, remainder = getopt.getopt(sys.argv[1:], 'o:v', ['access-token=', 'activities-file='])
    for opt, arg in options:
        if opt == '--access-token':
            access_token = arg
        elif opt == '--activities-file':
            activities_file = arg   
except:
    showUsage()
    exit(-1)

if access_token == None or activities_file == None:
    showUsage()
    exit(-1)

print "Access token    :", access_token
print "Activities file :", activities_file

# vk can hold the connection for a long time
socket.setdefaulttimeout(3)

# read lines with activities
dbFile = open(activities_file, 'r')
lines = dbFile.read().split("\n")
lines = list(set(lines))

# remove POST_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'POST_COMMENT ' in line:
        [type_id, owner_id, post_id, comment_id] = line.split(" ")
        if removeWallComment(access_token, owner_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "POST_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove POST
total_count = delete_count = 0
for line in lines:
    if 'POST ' in line:
        [type_id, owner_id, post_id] = line.split(" ")
        if removeWall(access_token, owner_id, post_id) == True:
            delete_count += 1
        total_count += 1
print "POST: %d of %d was removed" % (delete_count, total_count)

# remove PHOTO_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'PHOTO_COMMENT ' in line:
        [type_id, owner_id, comment_id] = line.split(" ")
        if removePhotoComment(access_token, owner_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "PHOTO_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove PHOTO
total_count = delete_count = edit_count = 0
for line in lines:
    if 'PHOTO ' in line:
        [type_id, owner_id, photo_id] = line.split(" ")
        if removePhoto(access_token, owner_id, photo_id) == True:
            delete_count += 1
        total_count += 1
print "PHOTO: %d of %d was removed" % (delete_count, total_count)

# remove VIDEO_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'VIDEO_COMMENT ' in line:
        [type_id, owner_id, video_id, comment_id] = line.split(" ")
        if removeVideoComment(access_token, owner_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "VIDEO_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove VIDEO
total_count = delete_count = edit_count = 0
for line in lines:
    if 'VIDEO ' in line:
        [type_id, owner_id, video_id] = line.split(" ")
        if removeVideo(access_token, owner_id, video_id) == True:
            delete_count += 1
        total_count += 1
print "VIDEO: %d of %d was removed" % (delete_count, total_count)

# remove TOPIC_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'TOPIC_COMMENT ' in line:
        [type_id, group_id, topic_id, comment_id] = line.split(" ")
        if removeTopicComment(access_token, group_id, topic_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "TOPIC_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove TOPIC
total_count = delete_count = edit_count = 0
for line in lines:
    if 'TOPIC ' in line:
        [type_id, group_id, topic_id] = line.split(" ")
        if removeTopic(access_token, group_id, topic_id) == True:
            delete_count += 1
        total_count += 1
print "TOPIC: %d of %d was removed" % (delete_count, total_count)


# remove NOTE_COMMENT
total_count = delete_count = 0
for line in lines:
    if 'NOTE_COMMENT ' in line:
        [type_id, user_id, note_id, comment_id] = line.split(" ")
        if removeNoteComment(access_token, user_id, comment_id) == True:
            delete_count += 1
        total_count += 1
print "NOTE_COMMENT: %d of %d was removed" % (delete_count, total_count)

# remove NOTE
total_count = delete_count = edit_count = 0
for line in lines:
    if 'NOTE ' in line:
        [type_id, user_id, note_id] = line.split(" ")
        if removeNote(access_token, note_id) == True:
            delete_count += 1
        total_count += 1
print "NOTE: %d of %d was removed" % (delete_count, total_count)
